File: Grind75/insert.py
def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
    new_start, new_end = newInterval
    result = []

    for interval in intervals:
        start, end = interval
        
        # No overlap, current interval comes before new interval
        if new_end < start:
            result.append([new_start, new_end])
            new_start, new_end = start, end
        # No overlap, current interval comes after new interval
        elif new_start > end:
            result.append(interval)
        else:
            new_start = min(new_start, start)
            new_end = max(new_end, end)
    
    result.append([new_start, new_end])
    return result

File: Grind75/insert.js
const insert = (intervals, newInterval) => {
  const result = [];
  let i = 0;
  const n = intervals.length;

  // Add all intervals before newInterval
  while (i < n && intervals[i][1] < newInterval[0]) {
    result.push(intervals[i]);
    i++;
  }

  // Merge overlapping intervals
  while (i < n && intervals[i][0] <= newInterval[1]) {
    newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
    newInterval[1] = Math.max(newInterval[1], intervals[i][1]);
    i++;
  }
  result.push(newInterval);

  // Add all intervals after newInterval
  while (i < n) {
    result.push(intervals[i]);
    i++;
  }

  return result;
};
